### ISSUE TITLE
Add more unit tests

### ISSUE THREAD

**Issue by Jaxan** [ocramz/sparse-linear-algebra#52]
Especially some basic ones. We should have (regression) tests for:
- [x] issue #42 
- [ ] issue #46 
- [x] issue #50 
- ... ?

**Comment by ocramz** [comment]
@Jaxan Yep, indeed. I will be travelling in the next two weeks but I'll try to make some time for this in April. Thanks for your help!
